lib/c-api/src/lib.rs

```diff
@@ -30,8 +30,51 @@
     unreachable_patterns
 )]
 
+use std::ffi::CString;
+use std::mem;
+use std::os::raw::c_char;
+
 #[cfg(feature = "deprecated")]
 pub mod deprecated;
 pub mod error;
 mod ordered_resolver;
 pub mod wasm_c_api;
+
+/// Gets the version of the Wasmer C API.
+///
+/// The returned string is guaranteed to be a valid C string. The
+/// caller owns the string, so it's up to the caller to free it.
+///
+/// # Example
+///
+/// ```rust
+/// # use inline_c::assert_c;
+/// # fn main() {
+/// #    (assert_c! {
+/// # #include "tests/wasmer_wasm.h"
+/// #
+/// int main() {
+///     // Read and print the version.
+///     char* version = wasmer_version();
+///     printf("%s", version);
+///
+///     // Free it, as we own it.
+///     free(version);
```

Comment:
From the docs:

> The pointer which this function returns must be returned to Rust and reconstituted using CString::from_raw to be properly deallocated. Specifically, one should not use the standard C free() function to deallocate this string.

I don't think we can ever mix C and Rust ownership: `CString` is an owned type on the Rust side, we can't pass it back to C without providing our own destructor. Probably simpler to have the user pass us a buffer or to return a pointer to a static string which does not need to be freed